target/linux/realtek/files-5.15/drivers/gpio/gpio-rtl8231.c

```diff
@@ -29,9 +29,10 @@ struct rtl8231_gpios {
 	int ext_gpio_indrt_access;
 };
 
-extern struct mutex smi_lock;
```

Comment:
While I'm all in favor for this improvement, I don't think it should be part of this changeset? I'd argue, do it in a small MR on its own. Also is this not being superseeded by @svanheule's rtl8231 mainline driver?

Reply:
You mean my out-of-tree rtl8231 driver, because it wasn't accepted upstream :-(

My RTL8231 driver also instatiates an MDIO device, so we would need support for the MDIO master first.

Reply:
I did mean, can you share the link to the rejection? Just curious as to what/why. From my pov; the chip sits on a bus, and exposes a GPIO interface. Just because that bus is an MDIO bus shouldn't matter (if everything is in place).

As for the MDIO bus, yes, I noticed that that is very messy right now, and was on pretty high on my todo lits :)

Reply:
not seeing a hard-rejection per say; but a (pretty big) technical challenge instead

Reply:
The driver itself was reviewed and could have been accepted, if it wasn't for those pesky little details in the kernel interface that I wanted (needed) to use...